File: Day4_Remove_K_Digits/RemoveKDigits.java
class Solution {
    public String removeKdigits(String num, int k) {
        Stack<Integer> stack=new Stack<>();
        for(int i=0;i<num.length();i++){
            while( !stack.isEmpty() && stack.peek()>
            (num.charAt(i)-'0') && k>0){
                stack.pop();
                k=k-1;
            }
            if( num.charAt(i)=='0' && stack.isEmpty()){
                continue;
            }
            stack.push(num.charAt(i)-'0');
            //System.out.println('B'-'A');
        }
        String ans="";
        while(k>0 && !stack.isEmpty()){
            stack.pop();
            k=k-1;
        }
        while(!stack.isEmpty()){
            ans=ans+(stack.pop());

        }
        
        if(ans.length()!=0){
            StringBuilder sb=new StringBuilder(ans);
            return sb.reverse().toString();
        }
        return "0";
    }
}

File: Day4_Remove_K_Digits/RemoveKDigits.py
class Solution(object):
    def removeKdigits(self, num, k):
        """
        :type num: str
        :type k: int
        :rtype: str
        """
        stack=[]
        for i in range(len(num)):
            while stack and stack[-1]>int(num[i]) and k>0: 
                stack.pop()
                k-=1
            if num[i]=='0' and not stack:
                continue
            stack.append(int(num[i]))
        ans=''
        while k>0 and stack:
            stack.pop()
            k-=1
        while stack:
            ans+=(str(stack.pop()))
        if ans:
            return ans[-1::-1]
        return '0'
            



        
